### PR TITLE
Disabling stuff that should not run when Deneb or Capella are not supported

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Maybe I missed something but it's all I caught disabling deneb constants. Also we should be careful with things like this in our code, it will not work as expected
```
forMilestone(highestSupportedMilestone)
        .getConfig()
        .toVersionDeneb()
        .map(SpecConfigDeneb::getMaxBlobsPerBlock);
```
It actually doesn't work like it should (being an Optional) because we have always some `SpecConfigDeneb`. 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java [hunk 1]
```diff
@@ -867,7 +871,12 @@ public boolean isAvailabilityOfBlobSidecarsRequiredAtEpoch(
   }
 
   public Optional<Integer> getMaxBlobsPerBlock() {
-    return forMilestone(getForkSchedule().getHighestSupportedMilestone())
+    final SpecMilestone highestSupportedMilestone =
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java:897]: what about

    final SpecMilestone highestSupportedMilestone =
      getForkSchedule().getHighestSupportedMilestone();
    return Optional.ofNullable(forMilestone(highestSupportedMilestone).getConfig())
        .flatMap(SpecConfig::toVersionDeneb)
        .map(SpecConfigDeneb::getMaxBlobsPerBlock);

File: beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/DefaultSyncServiceFactory.java [hunk 2]
```diff
@@ -127,9 +129,14 @@ public SyncService create(final EventChannels eventChannels) {
...
-    final FetchRecentBlobSidecarsService fetchRecentBlobSidecarsService =
-        FetchRecentBlobSidecarsService.create(
-            asyncRunner, blobSidecarPool, forwardSyncService, fetchTaskFactory, spec);
+    final RecentBlobSidecarFetcher fetchRecentBlobSidecarsService;
+    if (spec.isMilestoneSupported(SpecMilestone.DENEB)) {
+      fetchRecentBlobSidecarsService =
+          FetchRecentBlobSidecarsService.create(
+              asyncRunner, blobSidecarPool, forwardSyncService, fetchTaskFactory, spec);
+    } else {
+      fetchRecentBlobSidecarsService = RecentBlobSidecarFetcher.NOOP;
```
Comment (tbenr) [inline beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/DefaultSyncServiceFactory.java]: this naming difference is blowing up my mind.

**Review by tbenr — COMMENTED** [review]
in general looks good. But the only thing I'm not super convinced about is the general bjective. I dont like haveing all these ifs.
A part of me says we should minimize those spaghetti and enable what is safe to be enabled in advance.

**Comment by zilm13** [comment]
@tbenr did my best at refactoring this. Anyway I'm ok to NO MERGE if it's not needed

File: beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/ThrottlingSyncSource.java [hunk 3]
```diff
@@ -36,19 +37,21 @@ public class ThrottlingSyncSource implements SyncSource {
   private final SyncSource delegate;
 
   private final RateTracker blocksRateTracker;
-  private final RateTracker blobSidecarsRateTracker;
+  private final Optional<RateTracker> blobSidecarsRateTracker;
```
Comment (tbenr) [inline beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/ThrottlingSyncSource.java]: In this case I'd like to follow the `NOOP` pattern. Proliferation of Optionals is a form of spaghetti

**Review by tbenr — APPROVED** [review]
LGTM. I suggested one more improvement

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**